news/_posts/2020-12-09-pr-testing-on-prestashop.md

```diff
@@ -0,0 +1,182 @@
+Written specifications should be linked in the PR as well.
+
+All other information is the same.
+
+If some information is missing, the QA analyst will ping the author, the product team and/or the maintainers and wait for this information to be completed before starting to work on the PR.
+
+### Check that the improvement works accordingly to the issue and specification
+
+This step is simple: using the information in the issue, the "How to test" instructions, and the specification (if there is one), then QA analyst proceeds to check out the PR and try out each new behavior to make sure it works as expected and described.
+
+### Regression testing
+
+For improvements, the regression test part is even more important than for bug fixes: the QA analyst must make sure that the original features the PR improves are still working as expected.
+
+To do so, we generally compare two installs (one on the branch with the PR and one without), then perform the same actions on each install, and verify that the result is strictly identical. We can also carry out parts of the test campaign usually performed to verify build releases.
+
+### What happens next?
+
+Like with bug fixes, if the QA analyst finds any bug, they will comment the PR with all useful tips to help the author fix the bug.
+If the QA analyst notices an unspecified behavior (if the expected behavior is unclear, or for example, if it could be argued that an error should be displayed in a given scenario) they will ping the Product team and/or the maintainers to discuss this particular issue.
+
+If everything is fine, the QA analyst changes the label to “QA Approved” and the PR is merged by a maintainer, making it available on PrestaShop in the next build of the branch!
+
+Same as for bug fixes, these tests are tracked on an internal tool. The QA analyst might also update some of the test campaigns to add or modify test cases related to this improvement.
+
+# IV. Migration Pull Requests
+
+Migration Pull Requests are quite different to other ones: the purpose of these PRs is to move a page from the Legacy framework to the Symfony framework, while keeping the same features (and optionally adding some minor improvements).
+
+This makes the QA's work on these Pull Requests quite longer and harder compared to other PRs.
+
+### Requirements
+
+The main difference here is that the issue linked to the PR might be just an Epic listing all the items that should be migrated.
+
+For the “How to Test”, it’s sometimes only “Everything should work as in Legacy”. If there is a specification linked to this page, it should be added to the “How to Test”, to help the QA analyst make sure to check everything. If the migrated page is “hidden” (i.e. not accessible through the menu), the “How to Test” must also contain the URL to access it.
```

Comment:
```suggestion
For the “How to Test”, it’s sometimes only “Everything should work as in Legacy”. If there is a specification linked to this page, it should be added to the “How to Test”, to help the QA analyst make sure to check everything.
```
This part seems not necessary to me, that's a detail.

Reply:
I agree it's just a detail but it is often forgotten (and the majority of the migration PR needs this URL information).

Reply:
Yes but in my opinion it should be treated in the PR itself, not in an article. It's too much of a detail to be relevant here, even if I understand it's a big deal in the PR itself.
I'm just trying to remove some bits here and there to make the article more digest.
You are free to choose if you want to keep it or not 😃